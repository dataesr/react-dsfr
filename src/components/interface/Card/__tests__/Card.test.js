import { render, screen } from '@testing-library/react';
import Enzyme, { shallow } from 'enzyme';
import React from 'react';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import {
  Card, CardTitle, CardDescription, CardDetail, CardImage,
} from '..';

Enzyme.configure({ adapter: new Adapter() });

describe('<Card />', () => {
  it('renders Card properly', () => {
    render(
      <Card href="/" data-testid="card">
        <CardImage alt="alternative" src="http://fakeimg.pl/300/" data-testid="cardimage" />
        <CardDetail data-testid="carddetail">Détail</CardDetail>
        <CardTitle data-testid="cardtitle">
          Qu’est-ce que le Pass Culture et comment l’obtenir ?
        </CardTitle>
        <CardDescription data-testid="carddescription">
          Description texte body small regular consectetur adipisicing elit,
          sed do eiusmod tempor incididunt ut labore et dolore…
        </CardDescription>
      </Card>,
    );
    const card = screen.getByTestId('card');
    expect(card).toBeInTheDocument();
    expect(card).toMatchSnapshot();
  });
  it('renders horizontal Card properly', () => {
    render(
      <Card href="/" isHorizontal data-testid="card">
        <CardImage alt="alternative" src="http://fakeimg.pl/300/" />
        <CardDetail>Détail</CardDetail>
        <CardTitle>
          Qu’est-ce que le Pass Culture et comment l’obtenir ?
        </CardTitle>
        <CardDescription>
          Description texte body small regular consectetur adipisicing elit,
          sed do eiusmod tempor incididunt ut labore et dolore…
        </CardDescription>
      </Card>,
    );
    const card = screen.getByTestId('card');
    expect(card).toHaveClass('fr-card--horizontal');
  });
  it('renders Card class', () => {
    render(
      <Card data-testid="card" isGrey hasBorder={false} hasShadow>
        <CardImage alt="alternative" src="http://fakeimg.pl/300/" />
        <CardDetail>Détail</CardDetail>
        <CardTitle>
          Qu’est-ce que le Pass Culture et comment l’obtenir ?
        </CardTitle>
        <CardDescription>
          Description texte body small regular consectetur adipisicing elit,
          sed do eiusmod tempor incididunt ut labore et dolore…
        </CardDescription>
      </Card>,
    );
    const card = screen.getByTestId('card');
    expect(card.className).toEqual('fr-card fr-card--grey fr-card--shadow fr-card--no-border fr-card--sm');
  });

  it('Call Card onClick function', () => {
    const mockClick = jest.fn();
    const component = shallow(
      <Card onClick={mockClick}>
        <CardDetail>Détail</CardDetail>
        <CardTitle>
          Qu’est-ce que le Pass Culture et comment l’obtenir ?
        </CardTitle>
        <CardDescription>
          Description texte body small regular consectetur adipisicing elit,
          sed do eiusmod tempor incididunt ut labore et dolore…
        </CardDescription>
      </Card>,
    );

    component.simulate('click', { preventDefault: () => {} });
    expect(mockClick).toHaveBeenCalled();
  });
});
