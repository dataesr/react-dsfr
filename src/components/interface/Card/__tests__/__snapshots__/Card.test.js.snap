// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders Card properly 1`] = `
<div
  class="fr-card fr-enlarge-link"
  data-testid="card"
>
  <div
    class="fr-card__body"
  >
    <p
      class="fr-card__detail"
      data-testid="carddetail"
    >
      Détail
    </p>
    <p
      class="fr-card__title"
      data-testid="cardtitle"
    >
      <a
        class="fr-card__link"
        href="/"
        target="_self"
      >
        Qu’est-ce que le Pass Culture et comment l’obtenir ?
      </a>
    </p>
    <p
      class="fr-card__desc"
      data-testid="carddescription"
    >
      Description texte body small regular consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore…
    </p>
  </div>
  <div
    class="fr-card__img"
    data-testid="cardimage"
  >
    <img
      alt="alternative"
      src="http://fakeimg.pl/300/"
    />
  </div>
</div>
`;
